packages/amplify_datastore/android/src/test/kotlin/com/amazonaws/amplify/amplify_datastore/SchemaData.kt

```diff
@@ -170,6 +170,79 @@ val postAuthComplexSchema = ModelSchema.builder()
     .modelClass(SerializedModel::class.java)
     .build()
 
+val postAuthComplexWithProviderUserPoolsSchema = ModelSchema.builder()
+    .name("PostAuthComplexWithProviderUserPools")
+    .pluralName("PostAuthComplexesWithProviderUserPools")
+    .fields(
+        mapOf(
+            "id" to
+                    ModelField.builder()
+                        .name("id")
+                        .javaClassForValue(String::class.java)
+                        .targetType("String")
+                        .isRequired(true)
+                        .isArray(false)
+                        .build(),
+            "owner" to
+                    ModelField.builder()
+                        .name("owner")
+                        .javaClassForValue(String::class.java)
+                        .targetType("String")
+                        .isRequired(false)
+                        .isArray(false)
+                        .build()
+
+        )
+    )
+    .authRules(
+        listOf(
+            AuthRule.builder()
+                .authProvider(AuthStrategy.Provider.USER_POOLS)
+                .authStrategy(AuthStrategy.OWNER)
+                .ownerField("owner")
+                .identityClaim("cognito:username")
+                .build()
+        )
+    )
+    .modelClass(SerializedModel::class.java)
+    .build()
+
+val postAuthComplexWithProviderApiKeySchema = ModelSchema.builder()
+    .name("PostAuthComplexWithProviderApiKey")
+    .pluralName("PostAuthComplexesWithProviderApiKeys")
+    .fields(
+        mapOf(
+            "id" to
+                    ModelField.builder()
+                        .name("id")
+                        .javaClassForValue(String::class.java)
+                        .targetType("String")
+                        .isRequired(true)
+                        .isArray(false)
+                        .build(),
+            "owner" to
+                    ModelField.builder()
+                        .name("owner")
+                        .javaClassForValue(String::class.java)
+                        .targetType("String")
+                        .isRequired(false)
+                        .isArray(false)
+                        .build()
+
+        )
+    )
+    .authRules(
+        listOf(
+            AuthRule.builder()
+                .authProvider(AuthStrategy.Provider.API_KEY)
```

Comment:
Not 100% what this is testing but API key + owner Auth is not allowed by the CLI

Reply:
Ahh OK, I can update the test data.